src/sharing/ShareWrapper.js

```diff
@@ -178,19 +210,13 @@ class ShareWrapper extends React.Component<Props, State> {
       <>
         <ScrollView style={styles.wrapper} keyboardShouldPersistTaps="always" nestedScrollEnabled>
           <View style={styles.container}>
-            {sharedData.type === 'image' && (
-              <ImageBackground
-                source={{ uri: sharedData.sharedImageUrl }}
-                style={styles.imagePreview}
-              >
-                <Text style={styles.previewText}>{sharedData.fileName}</Text>
-              </ImageBackground>
-            )}
             {sharedData.type === 'file' && (
-              <View style={styles.imagePreview}>
-                <IconAttachment size={200} color={BRAND_COLOR} />
-                <Text style={styles.previewText}>{sharedData.fileName}</Text>
-              </View>
+              <FlatList
+                data={this.state.files}
+                renderItem={this.renderItem}
+                keyExtractor={i => i.url}
+                horizontal
```

Comment:
Horizontal sounds good here. Does it scroll if the previews don't fit in the width of the screen?

Reply:
Yes.

I just tested this, by temporarily setting the image width > my screen width, since images have fixed size of 200 x 200 which is smaller than most practical device widths and single image scrolls if its width exceeds screen width.

Reply:
Great. What if you have several images, so that each one fits but combined they don't?

Reply:
Then the flatlist scrolls horizontally